pt/scripts/iota_tracker.py

```diff
@@ -34,76 +34,154 @@
+        return map(tuple, zip(*db_datas))
+
+    # First query of the hour, no inserted data is normal
+    return [], []
+
+
+def get_decrypt_data_sign(receive_data):
+    # decrypt data
+    decrypt_data = PLAT_CIPHER.decrypt(
+        base64.b64decode(receive_data["data"]), RANDOM_GENERATOR
+    )
+
+    # check signature
+    # pylint: disable=E1102
+    is_verify = PLAT_SIGNER.verify(
+        SHA256.new(decrypt_data), base64.b64decode(receive_data["signature"])
+    )
+    # pylint: enable=E1102
+
+    return decrypt_data, is_verify
+
+
+def uniform_fields(insert_data, data_type):
+    insert_data["uuid"] = insert_data.pop("id")
+    if data_type == EV:
+        insert_data["power_display"] = insert_data.pop("power")
+    elif data_type == PV:
+        insert_data["pac"] = insert_data.pop("PAC")
+    elif data_type == WT:
+        insert_data["windgridpower"] = insert_data.pop("WindGridPower")
+    return insert_data
+
+
+def insert_to_db(tag, decrypt_data, db_uuid, address, receive_address):
+    data_type = IOTA_DATA_TYPE[tag[:-1]]
+    insert_data = json.loads(decrypt_data.decode())
+
+    # Prevent reinsertion of data using the same uuid
+    if insert_data["id"] in db_uuid:
+        logger.error(
+            f"Repeated Insert Error\n\
+            uuid: {insert_data['id']}\n\
+            address: {insert_data['address']}"
+        )
+        return
+
+    # parse day result from datetime.isoformat
+    # to process the time difference between IOTA and database
+    # python3.7+ can use datetime.fromisoformat(<isoformat>)
+    # try-catch is to prevent if `second` is an interger
+    try:
+        insert_data["updated_at"] = datetime.strptime(
+            insert_data["updated_at"], "%Y-%m-%dT%H:%M:%S.%f"
+        )
+    except ValueError:
+        insert_data["updated_at"] = datetime.strptime(
+            insert_data["updated_at"], "%Y-%m-%dT%H:%M:%S"
+        )
+
+    # get upload data's data_time
+    data_time = (
+        insert_data["updated_at"].astimezone(timezone.utc).replace(tzinfo=None).date()
+    )
+
+    # get related history id
+    history = History.query.filter_by(iota_address=address, time=data_time).first()
+    if history:
+        insert_data["history_id"] = history.uuid
+    else:
+        return
+
+    # put IOTA address into data_structure
+    insert_data["address"] = str(receive_address)
+
+    # Handling different names
+    insert_data = uniform_fields(insert_data, data_type)
+
+    # Insert data by ORM
+    try:
+        data_type(**insert_data).add()
+
+    # Check Insert same UUID's data
+    except sqlalchemy.exc.IntegrityError:
+        logger.error(
+            f"DB Insert Error: Unique Violation\n\
+            re-inserted data id: {insert_data['uuid']}\n\
+            address: {insert_data['address']}"
+        )
+
+
+def main(utc_now=datetime.utcnow()):
+    # get address and transaction hash from db
     addresses = [str(ami.iota_address) for ami in AMI.query.all()]
```

Comment:
If we are not going to change this object, we can use generator for better performance
```
addresses = (str(ami.iota_address) for ami in AMI.query.all())
```

Reply:
Fix in e95bbcc.